### DISCUSSION TITLE
Migrate to dotnet 8?

### DISCUSSION THREAD

**Discussion by honguyenminh** [seerge/g-helper#2704] [Q&A]
dotnet 7 reached end of life long ago, maybe it's time we move to dotnet 8? It's LTS too.

**Comment by seerge** [comment]
@honguyenminh i will eventually, but I need to find a way to do it gradually without forcing all 100 000 active users to download it and update :) 

There is nothing wrong with .NET 7 tho. Lots of apps use .NET6 or even .NET 3.5 / 4 still. It's just a library.

**Reply by sn4k3** [reply]
> I need to find a way to do it gradually without forcing all 100 000 active users

The only way is to publish the app as self-contained, it will include all net framework, still portable, but instead a single.exe you will have a ton of files. dotnet also gives the option to publish self-contained as single.exe (Large size, but AOT and trim is possible).
Other than that if you depend on system dotnet there's not way to do it "gradually", however the more time pass the more chances the user already have dotnet 8.0 installed, still majority won't. Now you may forcing 100 000 users, but latter you may force 200 000 users instead, with the annoyance of more users having residual net 7.0 installed which is not much popular (No new app should pick the dead net7 when net8 is here as LTS). IMO the transition is better soon than latter and will benefit everyone.

Also I see no problem in force your users to upgrade dotnet, if runtime not installed when running g-helper a popup will show and lead user to install the runtime, so even if they miss this new requirement, windows will take care of it:

![image](https://github.com/seerge/g-helper/assets/113281/39a3968a-65d7-410d-b39a-0c78be246259)


# Further notes

Quote from: https://devblogs.microsoft.com/dotnet/dotnet-7-end-of-support
.NET 7 will reach [end of support on May 14, 2024](https://github.com/dotnet/core/blob/main/release-notes/7.0/README.md). After that, Microsoft will n**o longer provide servicing updates, including security fixes** or technical support, for .NET 7. You’ll need to update to [.NET 8](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8/) before this date into stay supported.

May 14th is a patch Tuesday release day. .NET 7 may be updated one last time, on that day, if there is a known critical issue.

## You can expect the following after .NET 7 reaches end of support:

- Applications that use this version will continue to run.
- **No new security updates will be issued for .NET 7.**
- **Continuing to use an unsupported version will expose you to security vulnerabilities.**
- You may not be able to access technical support for .NET 7 applications.

## Using .NET 7 apps
- If you’re using a .NET 7 app, we recommend reaching out to the software developer or vendor who produced it to ask if an updated version that uses .NET 8 is available.

I think the critical point here is security, They say no more updates... I still believe if they find a big security hole they may fix it, but they don't have to... So apps should upgrade as soon as possible to avoid expose users to vulnerabilities. 
Devs should jump out of STS as soon LTS is out or before end of service. STS is never a "real" product in my POV.
